modules/OFStateManager/module/src/gentable_handlers.c

```diff
@@ -128,6 +129,15 @@ static indigo_core_gentable_t *gentables[MAX_GENTABLES];
  */
 static indigo_core_op_context_t op_ctx;
 
+/* outstanding op_ctx list for debug purpose */
+static list_head_t outstanding_op_ctx_list;
+
+static indigo_core_op_context_t * op_ctx_alloc(indigo_cxn_id_t cxn_id, of_object_t *obj);
+static void op_ctx_free(indigo_core_op_context_t *op_ctx);
```

Comment:
i don't think these prototypes are necessary because the functions are defined before they are used ; can you confirm?

Reply:
depend on the location of these function.

Reply:
i just looked at gentable_handlers.c, and it looks like op_ctx_alloc and op_ctx_free are not used before their definitions.  but if that is a concern, move the definitions to this point (e.g. starting at line 135) and you won't need the prototypes.